faust/assignor/partition_assignor.py

```diff
@@ -104,7 +105,8 @@ def _url(self) -> URL:
     def on_assignment(
             self, assignment: ConsumerProtocolMemberMetadata) -> None:
         metadata = cast(ClientMetadata,
-                        ClientMetadata.loads(assignment.user_data))
+                        ClientMetadata.loads(
+                            self._decompress(assignment.user_data)))
```

Comment:
Is this backwards compatible?

Should we use some magic header or something that we check for to see if the metadata is compressed?

Could be as simple as:

```py
COMPRESSED_MAGIC_HEADER = b'\x00\x0f\x03'

user_data = assignment.user_data
if user_data.startswith(COMPRESSED_MAGIC_HEADER):
   header = user_data[:4]
    _, _, _, compression_type = struct.unpack('>4b', header)
   
   # compression type 0 is zlib
   metadata = ClientMetadata.loads(self._decompress(user_data[4:], compression_type)
```

similarly to compress it:

```py
  @classmethod
     def _compress(cls, raw: bytes) -> bytes:
         return COMPRESSED_MAGIC_HEADER + struct.pack('>b', 0) + zlib.compress(raw)
```

Reply:
This should be backwards compatible as I have bumped up the assignor version. This would make all application instance to rejoin with fresh metadata eliminating issues due backwards incompatibility.